docs/architecture/clients/dependency-injection.md

```diff
@@ -0,0 +1,219 @@
+[dependency providers](https://angular.io/guide/dependency-injection-providers) to configure DI. For
+example:
+
+```ts
+@NgModule({
+  providers: [
+    // If the service does not have @Injectable decorators
+    {
+      provide: MyService, // abstract interface
+      useClass: DefaultMyService, // concrete implementation
+      deps: [DepA, DepB], // abstract interfaces for each dependency required by the DefaultMyService constructor
+    },
+    // If the service does have @Injectable decorators (Angular-only)
+    {
+      provide: SomeAngularService,
+      useClass: DefaultSomeAngularService,
+      // no deps array required
+    },
+  ],
+})
+export class MyServicesModule {}
+```
+
+By default, this is not type safe - Angular does not verify that the implementation actually
+implements the abstract interface, or that the `deps` array matches the constructor parameters.
+
+We provide a helper function called `safeProvider`, which acts as a wrapper around each provider
+entry and provides compile-time type checking of your configuration. For example (continuing with
+the example above):
+
+```ts
+@NgModule({
+  providers: [
+    safeProvider({
+      provide: MyService,
+      useClass: DefaultMyService,
+      deps: [DepA, DepB],
+    }),
+    safeProvider({
+      provide: SomeAngularService,
+      useClass: DefaultSomeAngularService,
+      useAngularDecorators: true, // this tells safeProvider to not require a deps array
+    }),
+  ],
+})
+export class MyServicesModule {}
+```
+
+:::note
+
+The `useAngularDecorators` option is not type safe - it is your responsibility to ensure that the
+class actually uses the `@Injectable` decorator.
+
+:::
+
+Our primary service modules require the use of `safeProvider`, otherwise you will receive type
+errors. However, you should use `safeProvider` wherever you are configuring an Angular provider to
+benefit from its type checking.
+
+### Angular root module
+
+Services that will be used globally and which don't have an abstract interface can be registered
+using the `provideIn` property. This avoids having to create and manage service modules. When
+exporting services through feature modules, it can be tricky to manage their lifetime, and it's easy
+to end up with instances of the service being created.
+
+```ts
+@Injectable({
+  provideIn: "root",
+})
+export class MyService {}
+```
+
+### Angular components
+
+Services that do are not stateful and/or are only supposed to be used locally to extract complex
+logic from a component can be injected directly into standalone components, completely bypassing
+modules.
+
+```ts
+@Component({
+  providers: [
+    // This example assumes that MyService uses @Injectable decorators and does not have an abstract interface
+    safeProvider(MyService),
+  ],
+  standalone: true,
+})
+export class MyComponent {}
+```
+
+### Non-Angular contexts
+
+Non-Angular contexts do not use a DI framework and must manually instantiate their dependencies **in
```

Comment:
typo?: 
```suggestion
Non-Angular contexts do not use a DI framework must manually instantiate their dependencies **in
```

Reply:
Nope! It's saying 2 things:
1. Non-Angular contexts do not use a DI framework
2. Non-Angular contexts must manually instantiate their dependencies

To put it another way, the "and" is a logical "and therefore".

Reply:
Wow, how did I not get that :D Maybe add "therefore" just in case?

:art: suggestion(non-blocking): 
```suggestion
Non-Angular contexts do not use a DI framework and must therefore manually instantiate their dependencies **in
```